### PR TITLE
[ENH] Refactor `CutoffSplitter` using `get_window` function

### PR DESCRIPTION
**Pull request by khrapovs**
#### Reference Issues/PRs

Next step in #1737 

#### What does this implement/fix? Explain your changes.

This PR is a simple refactoring. Use `sktime.datatypes._utilities.get_window` in `sktime.forecasting.model_selection._split.CutoffSplitter._split`.

#### Does your contribution introduce a new dependency? If yes, which one?

No

##### For all contributions
- [X] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [X] I've added unit tests and made sure they pass locally.
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

### CODE REVIEW

**Comment by fkiraly** [comment]
hm, something is broken here? tests fail

**Comment by khrapovs** [comment]
While trying to make this PR work I noticed these lines in `sktime.forecasting.base._sktime._BaseWindowForecaster._predict_in_sample`:
```
cutoffs = fh.to_relative(self.cutoff) + len(y_train) - 2
cv = CutoffSplitter(cutoffs, fh=1, window_length=self.window_length_)
```
This results in `cutoffs = [-1, 0, 1, 2, ...]`. If integer cutoffs are treated as `iloc` references, then `-1` is the reference to the very last element in `y_train` which drives `CutoffSplitter` crazy. I have treated this case in the splitter separately, but it seems to me we have a bigger problem here.

**Comment by fkiraly** [comment]
>but it seems to me we have a bigger problem here.

Could this be the source of the problems with `update_predict` tests that creates empty folds?

**Comment by khrapovs** [comment]
> Could this be the source of the problems with `update_predict` tests that creates empty folds?

I am not aware of these problems, but it sounds very much related. `cutoff = -1` results in the empty `train_window = np.array([])`. For whatever historical reasons this is ok for the splitters, it is a tested outcome. But I still struggle to understand this.

**Comment by khrapovs** [comment]
Just for a fun of it I tried to remove negative cutoffs like that:
```
cutoffs = fh.to_relative(self.cutoff) + len(y_train) - 2
cv = CutoffSplitter(cutoffs[cutoffs >= 0], fh=1, window_length=self.window_length_)
```
but this resulted in failure of some estimators. I decided to localise the changes inside of splitters.

**Comment by fkiraly** [comment]
I am getting very confused.

My typical fallback in this case is the docstring, which imo is not detailed enough.
Could you kindly add your understanding there what the splitter does:

* let's start with a full index set $t_1, ..., t_N$
* and an input of cutoffs $k_i$ and a forecasting horizon indices $h_i$
* what is the set of training indices output? what is the set of test indices output precisely?
* for instance, line 682, it should be the test set, not the training set, right?

**Comment by fkiraly** [comment]
> Just for a fun of it I tried to remove negative cutoffs like that

Not to judge, but I can imagine 100 things that seem more fun, such as cutting my own nose off, or eating a can of nails

**Comment by khrapovs** [comment]
I have updated the docstring with train_window math definition. 

> * for instance, line 682, it should be the test set, not the training set, right?

Yes, changed.
